managers/views.py

```diff
@@ -89,17 +101,31 @@ def get_context_data(self, **kwargs: Any) -> dict:
         return context_data
 
     def get_success_url(self) -> str:
+        logger.info(f"User with id: {self.request.user.pk} is in ProjectUpdate view for project with id: {self.kwargs['pk']}")
         return reverse("custom-project-detail", kwargs={"pk": self.kwargs["pk"]})
 
+    def form_valid(self, form: ProjectForm) -> HttpRequest:
+        project = super().form_valid(form)
+        logger.info(f"Project with id: {project.pk} has been updated by user with id: {self.request.user.pk}")
+        return project
+
 
 class ProjectDeleteView(DeleteView):
     model = Project
 
     def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         if not request.user.is_admin:
+            logger.debug(f"User with id: {request.user.pk} want to delete project")
             return redirect(reverse("home"))
 
         return super().dispatch(request, *args, **kwargs)
 
     def get_success_url(self) -> str:
         return reverse("custom-projects-list")
+
+    def delete(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        project = super().get_object()
```

Comment:
Use `super().delete()` here too instead of rewriting logic.

Reply:
done